datajunction-server/tests/api/djql_test.py

```diff
@@ -0,0 +1,380 @@
+
+    response = client_with_query_service.get(
+        "/djql/data/",
+        params={"query": query},
+    )
+    assert (
+        response.json()["message"]
+        == "HAVING, LATERAL VIEWS, and SET OPERATIONS are not allowed on `metrics` queries."
+    )
+
+
+def test_get_djql_illegal_column_expression(
+    client_with_query_service: TestClient,
+) -> None:
+    """
+    Test djql with non col exp in projection
+    """
+
+    query = """
+        SELECT
+        default.avg_repair_price avg_repair_price,
+        default.hard_hat.country,
+        default.hard_hat.id+5
+        FROM
+        metrics
+        GROUP BY
+        default.hard_hat.country,
+        default.hard_hat.city
+    """
+
+    response = client_with_query_service.get(
+        "/djql/data/",
+        params={"query": query},
+    )
+    assert (
+        response.json()["message"]
+        == "Only direct Columns are allowed in `metrics` queries, found `default.hard_hat.id + 5`."
+    )
+
+
+def test_get_djql_illegal_column(
+    client_with_query_service: TestClient,
+) -> None:
+    """
+    Test djql with bad col in projection
+    """
+
+    query = """
+        SELECT
+        default.avg_repair_price avg_repair_price,
+        default.hard_hat.country,
+        default.repair_orders.id
+        FROM
+        metrics
+        GROUP BY
+        default.hard_hat.country,
+        default.hard_hat.city
+    """
+
+    response = client_with_query_service.get(
+        "/djql/data/",
+        params={"query": query},
+    )
+    assert (
+        response.json()["message"]
+        == "You can only select direct METRIC nodes or a column from your GROUP BY on `metrics` queries, found `default.repair_orders.id`"
+    )
+
+
+def test_get_djql_illegal_limit(
+    client_with_query_service: TestClient,
+) -> None:
+    """
+    Test djql with bad limit
+    """
+
+    query = """
+        SELECT
+        default.avg_repair_price avg_repair_price
+        FROM
+        metrics
+        GROUP BY
+        default.hard_hat.country
+        LIMIT 1+2
+    """
+
+    response = client_with_query_service.get(
+        "/djql/data/",
+        params={"query": query},
+    )
+    assert (
+        response.json()["message"]
+        == "LIMITs on `metrics` queries can only be integers not `1 + 2`."
+    )
+
+
+def test_get_djql_no_nodes(
+    client_with_query_service: TestClient,
+) -> None:
+    """
+    Test djql without dj node refs
+    """
+
+    query = """
+        SELECT 1
+    """
+
+    response = client_with_query_service.get(
+        "/djql/data/",
+        params={"query": query},
+    )
+    assert response.json()["message"].startswith("Found no dj nodes in query")
+
+
+def test_djql_stream(
+    client_with_query_service: TestClient,
+) -> None:
+    """
+    Test streaming djql
+    """
+    query = """
+    SELECT default.hard_hat.country,
+    default.hard_hat.city
+    FROM default.hard_hat
+        """
+
+    response = client_with_query_service.get(
```

Comment:
I believe this test is failing in the CI because it's missing a few arguments. This should work:
```py
response = client_with_query_service.get(
    "/djql/stream/",
    params={"query": query},
    headers={
        "Accept": "text/event-stream",
    },
    stream=True,
)
```